test/unit/lib/TopStoriesFeed.test.js

```diff
@@ -91,12 +91,86 @@ describe("Top Stories Feed", () => {
+    it("should bind parseOptions to SectionsManager.onceInitialized when discovery stream is true", () => {
+      instance.onAction({type: at.PREFS_INITIAL_VALUES, data: {"discoverystream.config": JSON.stringify({enabled: true})}});
+      assert.calledOnce(sectionsManagerStub.onceInitialized);
+    });
+    it("should bind parseOptions to SectionsManager.onceInitialized when discovery stream is false", () => {
+      instance.onAction({type: at.PREFS_INITIAL_VALUES, data: {"discoverystream.config": JSON.stringify({enabled: false})}});
+      assert.calledOnce(sectionsManagerStub.onceInitialized);
+    });
+
+    it("should not re-initialize properties", () => {
+      // For discovery stream experience disabled TopStoriesFeed properties
+      // are initialized in constructor and should not be called again while lazy loading topstories
+      instance.initializeProperties = sinon.spy();
+      instance.isInitialized = true;
+      instance.lazyLoadTopStories();
+      assert.notCalled(instance.initializeProperties);
+    });
+
+    it("should have early exit onInit when discovery is true", async () => {
+      instance.discoveryStreamEnabled = true;
+      instance.doContentUpdate = sinon.spy();
+      await instance.onInit();
+      assert.notCalled(instance.doContentUpdate);
+      assert.isUndefined(instance.storiesLoaded);
+    });
+
+    it("should complete onInit when discovery is false", async () => {
+      instance.discoveryStreamEnabled = false;
+      instance.doContentUpdate = sinon.spy();
+      await instance.onInit();
+      assert.calledOnce(instance.doContentUpdate);
+      assert.isTrue(instance.storiesLoaded);
+    });
+
+    it("should handle limited actions when discoverystream is enabled", async () => {
+      instance.discoveryStreamEnabled = true;
+      instance.handleDisabled = sinon.spy();
+      instance.getPocketState = sinon.spy();
+      instance.onAction({type: at.PREFS_INITIAL_VALUES, data: {"discoverystream.config": JSON.stringify({enabled: true})}});
+      assert.calledOnce(instance.handleDisabled);
+
+      instance.onAction({type: at.NEW_TAB_REHYDRATED, meta: {fromTarget: {}}});
+      assert.notCalled(instance.getPocketState);
+    });
+
+    it("should handle NEW_TAB_REHYDRATED when discoverystream is disabled", async () => {
+      instance.discoveryStreamEnabled = false;
+      instance.handleDisabled = sinon.spy();
+      instance.getPocketState = sinon.spy();
+      instance.onAction({type: at.PREFS_INITIAL_VALUES, data: {"discoverystream.config": JSON.stringify({enabled: false})}});
+      assert.notCalled(instance.handleDisabled);
+
+      instance.onAction({type: at.NEW_TAB_REHYDRATED, meta: {fromTarget: {}}});
+      assert.calledOnce(instance.getPocketState);
+    });
+
+    it("should fire init on PREF_CHANGED", () => {
+      instance.onInit = sinon.spy();
+      instance.onAction({type: at.PREF_CHANGED, data: {name: "discoverystream.config", value: {}}});
```

Comment:
This test isn't testing with `discoveryStreamEnabled = false`

Reply:
Or true I think we actually want. Perhaps we want to put `instance.discoveryStreamEnabled = true;` in a `beforeEach` and only explicitly set it to false in certain tests.

Reply:
Done